docs/architecture/grpc-for-wcf-developers/authentication.md

```diff
@@ -1,11 +1,42 @@
 ---
-title: gRPC authentication and authorization - gRPC for WCF developers
-description: TO BE WRITTEN
+title: Security in gRPC applications - gRPC for WCF developers
+description: Overview of call and channel authentication and authorization in gRPC
 author: markrendle
 ms.date: 09/02/2019
 ---
 
-# gRPC authentication and authorization
+# Security in gRPC applications
+
+In any real-world scenario, securing applications and services is essential. Security covers two key areas: encrypting network traffic to prevent it from being intercepted by bad actors; and authenticating users to prevent unauthorized system access and apply permissions based on identity. This chapter will cover the facilities for authentication and authorization in gRPC for ASP.NET Core, and discuss network security using TLS encrypted connections.
+
+## WCF authentication
+
+In WCF, authentication and authorization was handled in different ways depending on the transports and bindings being used. WCF supported various WS-* security standards, as well as Windows authentication for HTTP services running in IIS or NetTCP services between Windows systems.
```

Comment:
WCF really only has two ways to handle authorization. Either a custom AuthorizationManager or via some attributes on the service implementation. Configuring authentication is done on the binding. If you use transport security, the authentication is usually configured via Binding.Security.Transport. If you use a custom binding, then NetTcp requires adding of specific binding elements such as SslStreamSecurityBindingElement or WindowsStreamSecurityBindingElement. Encryption can be tied to this, e.g. for NetTcp the client credential type controls whether you use TLS or NegotiateStream. HTTPS only uses TLS. With message security, there are lots of ways to configure the authentication and encryption.

Reply:
I don't think an in-depth discussion of how WCF actually handles authorization is necessary in this context.

Reply:
I was giving you an overview so you could choose which things you wanted to talk about. For example, Saying something about if you are using a custom AuthorizationManager with WCF, then you would implement that using the authorization mechanism in Asp.NET Core.

---
In reply to: [326138614](https://github.com/dotnet/docs/pull/14307#discussion_r326138614) [](ancestors = 326138614)